include/boost/geometry/algorithms/detail/overlay/get_turn_info.hpp

```diff
@@ -1191,139 +1369,136 @@ struct get_turn_info
 
         char const method = inters.d_info().how;
 
+        if (method == 'd')
+        {
+            // Disjoint
+            return out;
+        }
+
         // Copy, to copy possibly extended fields
         TurnInfo tp = tp_model;
 
-        bool do_only_convert = false;
+        bool const handle_as_touch_interior = method == 'm';
+        bool const handle_as_cross = method == 'i';
+        bool handle_as_touch = method == 't';
+        bool handle_as_equal = method == 'e';
+        bool const handle_as_collinear = method == 'c';
+        bool const handle_as_degenerate = method == '0';
```

Comment:
This block is changed from `switch` to boolean variables and `if`, mainly because in some cases first the `touch_interior` is tried but then it falls back to a `touch`, and `switch` is not convenient for that.